### DISCUSSION TITLE
Column, index name length validation

### DISCUSSION THREAD

**Discussion by fairking** [fluentmigrator/fluentmigrator#1523] [Q&A]
Is there any way to make a validation so all columns, indexes and keys cannot exceed the particular length (in my case 30 characters)?

**Comment by jzabroski** [comment]
Probably.  In theory, it should just use the same infrastructure @fubar-coder added for convention sets, to validate people stay within the convention they define.  See: `IMigrationValidator` example in https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/test/FluentMigrator.Tests/MigrationValidatorTests.cs

I have not given it a go, but it seems like the API is there.  You would need to use a type-switch pattern to dispatch on the "expression language" FluentMigrator supports, e.g. all IMigrationExpression "roots" (abstract base classes)

I would recommend using the FluentValidation library for validating properties.  It's a pretty nice library that makes it easy to write tests for validation rules, as well as expression validation rules.

**Reply by fairking** [reply]
>>  See: IMigrationValidator example

Looks like something very complicated ;-( I do not understand how can I get it work.

Is there any naming convention I can attach my validation to? I cannot see any info on the doc page: https://fluentmigrator.github.io/articles/intro.html
